src/App.jsx

```diff
@@ -14,33 +16,101 @@ export const goodsFromServer = [
-      </button>
+const SORT_ALPHABETICALLY = 'alphabetically';
+const SORT_BY_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getSortedGoods(goods, sortGoods, isReversed) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortGoods) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortGoods) {
+        case SORT_ALPHABETICALLY:
+          return good1.localeCompare(good2);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortGoods, setSortGoods] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const sortedGoods = getSortedGoods(goodsFromServer, sortGoods, isReversed);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => {
+            setSortGoods(SORT_ALPHABETICALLY);
+            setIsReversed(false);
+          }}
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortGoods !== SORT_ALPHABETICALLY || isReversed,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => {
+            setSortGoods(SORT_BY_LENGTH);
+            setIsReversed(false);
+          }}
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortGoods !== SORT_BY_LENGTH || isReversed,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => {
+            setIsReversed(!isReversed);
+          }}
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !isReversed,
+          })}
+        >
+          Reverse
+        </button>
+
+        {(sortGoods !== '' || isReversed) && (
+          <button
+            onClick={() => {
+              setSortGoods('');
+              setIsReversed(false);
+            }}
+            type="button"
+            className={cn('button is-danger', {
+              'is-light': sortGoods !== '' || isReversed,
+            })}
+          >
+            Reset
+          </button>
+        )}
+      </div>
+
+      <ul>
+        {sortedGoods.map(good => (
+          <li data-cy="Good">{good}</li>
```

Comment:
The `data-cy` attribute is used for testing purposes, typically with Cypress. Ensure that this attribute is correctly used in your testing setup. If it's not needed for your tests, consider removing it to clean up the code.